software/base/src/main/java/brooklyn/entity/software/MachineLifecycleEffectorTasks.java

```diff
@@ -155,29 +155,42 @@ public void attachLifecycleEffectors(Entity entity) {
                 .build();
     }
 
+    /** @see {@link #newStartEffector()} */
+    public Effector<Void> newSuspendEffector() {
+        return Effectors.effector(Void.class, "suspend")
+                .description("Suspend the process/service represented by an entity")
+                .parameter(StopSoftwareParameters.STOP_PROCESS_MODE)
+                .parameter(StopSoftwareParameters.STOP_MACHINE_MODE)
+                .impl(newSuspendEffectorTask())
+                .build();
+    }
+
     /**
-     * Returns the {@link TaskFactory} which supplies the implementation for the start effector.
+     * Returns the {@link EffectorBody} which supplies the implementation for the start effector.
      * <p>
      * Calls {@link #start(Collection)} in this class.
      */
     public EffectorBody<Void> newStartEffectorTask() {
-        return new EffectorBody<Void>() {
-            @Override
-            public Void call(ConfigBag parameters) {
-                Collection<? extends Location> locations  = null;
+        return new StartEffectorBody();
```

Comment:
This will break rebinding as the anonymous classes will be needed when reconstructing the entities. The approach used to keep backwards compatibility in similar cases is to keep the methods creating the anonymous classes but not use them, i.e. rename to newStartEffectorTaskDeprecated and change its visibility to private. Only needed for the newXXXEffectorTask methods. Once apache#738 is merged one could instead add code to do the rename on the fly.
Apologies for writing this late, was just able to review this.